doc/asset-contracts-cbor.md

```diff
@@ -0,0 +1,189 @@
+  * Elements Core and any software depending on it, such as block explorers, will consider such assets "unknown".
+  * The Liquid Asset Registry will reject such assets.
+  * It will still be possible to distribute such asset contracts onchain as long as they are under the maximum size, but compliant consumers may reject them.
+
+The intent of the restrictions described above is to simplify safe parsing of contracts, without making contract creation difficult for issuers.
+* Issuers who want maximum compatibilty for their contracts, including use of the asset registry or Elements Core, MUST ensure their contracts are well-formed.
+  * The described subset of CBOR is selected for maximum compatibility; the excluded features are obscure, and users are unlikely to run into any of them by accident.
+  * Issuers who use the asset registry can validate their contract against this specification using the registry contract validation API.
+  * Other tooling will be available for local validation of contracts, including Elements Core RPC.
+* Consumers have some flexibility in parsing contracts.
+  * Consumers MUST NOT crash or exhibit exploitable behavior when encountering an invalid contract. Consumers who get contracts from an untrusted source (such as the blockchain) MUST ensure that their code, including third-party code such as a CBOR parser, does not crash or exhibit exploitable behavior regardless of input, including invalid CBOR, or valid CBOR representing an invalid asset contract. Malicious parties can easily put arbitrary data on the blockchain where a contract is expected.
+  * Either accepting or rejecting an invalid contract is in compliance with this specification; only crashes and exploitable behavior are prohibited.
+  * Consumers who trust and rely on an asset registry MAY assume that version 1 contracts downloaded from the registry are well-formed.
+  * Consumers who get their contracts from an untrusted source SHOULD refuse to use assets whose contracts are not well-formed, but are not required to do so.
+  * Consumers (such as hardware wallets) who wish to minimize parsing complexity may rely on the fact that a valid version 1 contract will always begin as follows: `0x01 0x83 <1 byte precision, value 0-8> 0x73-0x75 <3-5 byte ticker, ASCII text> ...`
+    * Of course, such consumers MUST NOT crash or display exploitable behavior if a contract received from an untrusted source turns out to be invalid.
+
+Registry rules:
+* A registry may impose additional restrictions on contracts, beyond those described above.
+* In particular: The Liquid Asset Registry requires the `name`, `issuer_pubkey`, and `domain` fields to be present, and the `domain` field must be controlled by the issuer, which must be proven by the existence of a specific file served over https from that domain.
+* The domain validation rule helps protect registry users against assets which are intended to spoof other assets.
```

Comment:
What is the domain validation rule exactly ?

Reply:
Added more info!